Added KeySpace Integration

cloudonshore opened the pull request:
This adds integration with KeySpace encryption features. Encrypted messages sent to the API server will now be automatically decrypted (given that the API server has run the `/registerPGPKey` function).

grahamperich commented on api-server/lib/AirSwap.js:
```diff
@@ -456,6 +474,77 @@ class AirSwap {
+    } else {
+      const ipfsHash = await this.pgpContract.addressToPublicKey(this.wallet.address)
+      this.pgpKeys[address] = await ipfs.cat(ipfsHash)
+      return this.pgpKeys[address]
```
this can fail if wallet address is not registered right? should we return an error in that case?

cloudonshore replied:
Yes, changed. In terms of communicating that to whatever is calling the API server, I'm not sure how we should be handling that in general. Probably should use error codes (like responses with status `500` when an error occurs, instead of the `200` we currently always send). But that's outside of the scope of this PR.

grahamperich commented on api-server/server.js:
```diff
@@ -144,5 +147,21 @@ app.post(
   }),
 )
 
+app.post(
```
please also document these in `docs/serverAPI/api.md`

cloudonshore replied:
done

cloudonshore commented:
Cleaned up a few things, removed yarn.lock files since the documentation specifies npm install instead of yarn, and having both yarn.lock and package-lock can cause non-deterministic behavior.
